src/librustc_codegen_ssa/mir/block.rs

```diff
@@ -770,6 +770,18 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 &fn_abi.args[first_args.len()..])
         }
 
+        let needs_location =
+            instance.map(|i| i.def.requires_caller_location(self.cx.tcx())).unwrap_or_default();
+        if needs_location {
+            assert_eq!(
+                fn_abi.args.len(), args.len() + 1,
+                "#[track_caller] fn's must have 1 more argument in their ABI than in their MIR",
+            );
+            let location = self.get_caller_location(&mut bx, span);
+            let last_arg = &fn_abi.args.last().unwrap();
```

Comment:
I've seen this in a couple places, but `last` returns `Option<&T>`, so you're creating a `&&T`.
It works because of deref coercions, but you can just remove the `&`.